theories/ring.elpi

```diff
@@ -9,6 +9,26 @@ ring->field R (some F) :-
   coq.unify-eq {{ GRing.Ring.sort lp:R }} {{ GRing.Field.sort lp:F }} ok, !.
 ring->field _ none.
 
+pred field->unitRing o:term, o:term.
+field->unitRing F R :- !,
+  coq.unify-eq {{ GRing_Field__to__GRing_UnitRing lp:F }} R ok.
+
+pred unitRing->ring o:term, o:term.
+unitRing->ring UR R :- !,
+  coq.unify-eq {{ GRing_UnitRing__to__GRing_Ring lp:UR }} R ok.
+
+pred ring->sring o:term, o:term.
+ring->sring R SR :- !, coq.unify-eq {{ GRing_Ring__to__GRing_SemiRing lp:R }} SR ok.
+
+pred ring->zmod o:term, o:term.
+ring->zmod R V :- !, coq.unify-eq {{ GRing_Ring__to__GRing_Zmodule lp:R }} V ok.
+
+pred ring->nmod o:term, o:term.
+ring->nmod R V :- !, coq.unify-eq {{ GRing_Ring__to__GRing_Nmodule lp:R }} V ok.
+
+pred zmod->nmod o:term, o:term.
+zmod->nmod U V :- !, coq.unify-eq {{ GRing_Zmodule__to__GRing_Nmodule lp:U }} V ok.
```

Comment:
We should probably factor that in `common.elpi` tather than duplicating it in `ring.elpi` and `lra.elpi`